src/poetry/installation/chooser.py

```diff
@@ -177,6 +177,9 @@ def _no_links_found_error(
                 f"Make sure the lockfile is up-to-date. You can try one of the following;\n\n"
                 f"    1. <b>Regenerate lockfile: </><fg=yellow>poetry lock --no-cache --regenerate</>\n"
                 f"    2. <b>Update package     : </><fg=yellow>poetry update --no-cache {package.name}</>\n\n"
+                # FIXME: In the future, it would be better to suggest a more targeted
+                # cache clear command for just the package in question, tracked by https://github.com/python-poetry/poetry/issues/4438
+                f"If any of those solutions worked, you will have to clear your caches using (<c1>poetry cache clear --all .</>).\n\n"
```

Comment:
Is there a way to find which cache the package was pulled from?

It would be nice to specify the exact cache name to clear.

I don't think it's very useful to just say `CACHE_NAME` as then you need to figure out which cache the package belongs to. Which means the user needs to find the `poetry cache list` command, then make some leaps/assumptions on which cache contains the package. ex. `poetry cache clear --all PyPI`

Reply:
I tried `package.source_reference` but that ends up as `None` for me locally.

More context: `f"poetry cache clear {package.source_reference}:{package.name}:{package.version}"` -> `poetry cache clear None:matrix-synapse:1.140.0`

Reply:
That is the logic to find the source:

https://github.com/python-poetry/poetry/blob/6ace575663d3fd5541793f32b602c0b4afd48373/src/poetry/installation/chooser.py#L194-L201

If it is None, then it is either PyPI or the first repository if PyPI is disabled. (Repository names are case-insensitive.)